lib/grovepi/pivotpi/PCA9686.ex

```diff
@@ -0,0 +1,78 @@
+defmodule GrovePi.PivotPi.PCA9685 do
+  @moduledoc false
+
+  # References
+  # https://github.com/DexterInd/PivotPi/tree/master/Software/Python
+  # https://www.nxp.com/docs/en/data-sheet/PCA9685.pdf
+
+  # registers/etc:
+  @pca9685_address     0x40
+  @mode1               0x00
+  @mode2               0x01
+  # @subadr1             0x02 # Unused
+  # @subadr2             0x03 # Unused
+  # @subadr3             0x04 # Unused
+  @prescale            0xfe
+  @led0_on_l           0x06
+  @led0_on_h           0x07
+  @led0_off_l          0x08
+  @led0_off_h          0x09
+  @all_led_on_l        0xfa
+  @all_led_on_h        0xfb
+  @all_led_off_l       0xfc
+  @all_led_off_h       0xfd
+
+  # bits:
+  @allcall             0x01
+  @outdrv              0x04
+  @sleep               0x10
+  @invrt               0x10
+  # @restart             0x80 # Unused
+
+  @default_freq        60
+  @default_mode1       0x01
+
+  alias GrovePi.{Board}
+  import Bitwise
+
+  def start() do
+    set_all_pwm(0, 0)
+    set_initial_config()
+    set_pwm_freq(@default_freq)
+  end
+
+  defp set_initial_config() do
+    send_cmd(<<@mode2, (@outdrv ||| @invrt)>>) # Totem pole drive, and inverted signal.
+    send_cmd(<<@mode1, @allcall>>)
+    Process.sleep(50)
```

Comment:
Do you know why this sleep and the couple sleeps below are needed?

Reply:
I think the first one can be removed.  For the next one, it takes 500 microseconds max for the oscillator to be up and running once SLEEP bit has been set to logic 0.  See note 2 on pg 14.  The default_mode1 sets the SLEEP bit logic to 0.

https://www.nxp.com/docs/en/data-sheet/PCA9685.pdf 

The first one is there because the Python program cycles through sleep mode during initialization, but I couldn't find where that was necessary.  I forgot to also delete the sleep.

Reply:
I removed the extra sleep. Now it is only called after the device is given the wake command with `@default_mode1`.  I also reduced it to 1 millisecond which covers the 500 microsecond period.